### ISSUE TITLE
Disambiguation Page of All Notes

### ISSUE THREAD

**Issue by SpaceForever** [foambubble/foam#421]
This is a proposal for a feature that creates a page of all your markdown notes, so that people navigating your Foam site will be able to locate pages that would be otherwise orphaned.

**Comment by riccardoferretti** [comment]
do I understand correctly that this is an index page for all your notes?

**Comment by rajlego** [comment]
Yes, that's what we have in mind. It's difficult to find a lot of notes (for browsers of our site) since many are orphans (since wiki is still relatively new) and others might be hidden more deeply.

**Comment by MCluck90** [comment]
I'd like to make a request maybe for a v2 of this feature: allow the user to specify some sort of filter. In my case, I would find value in indexing most of my notes but indexing all of my daily notes would just create a lot of noise.

**Comment by rajlego** [comment]
Small extra request for a maybe v3...could you sort by 'weight'? Put things with most links to/from at the top and descend?
I have a burning hatred of sites like [Paul Graham's](http://www.paulgraham.com/index.html) which have amazing content but force me to go through them in random, arbitrary inefficient order

**Comment by Koleok** [comment]
Pretty easy to just write a script for this and run it prepush or something, I initially did just that when I started my repo, but found that it rendered the graph much less useful with hundreds of useless edges that don't represent conceptual connections. In order for a master index to be useful it'd be nice to be able to tell the graph to ignore specific files, in a foam config file or something.

**Comment by Koleok** [comment]
I might take a crack at this actually, nice easy entry issue 👌

**Comment by daniel-vera-g** [comment]
I recently migrated my "regular" notes to foam and created "local" index files that link to related concepts. I used a simple shell script.

---

For this use case, I would used a simple `find` command that searches all local markdown files and inserts them in the index file with the foam link-syntax.  
- I use [fd](https://github.com/sharkdp/fd):  `fd -e md -E '*/journal/*' -x echo '- [['{/.}']]' > index.md`
- But you can also achieve same results with find(I don't use it regulary...maybe there's a simpler option): 
`find  ./ -name '*.md' -not -path '*/journal/*' -exec basename {} .md \; | while read filename;do echo '- [['$filename']]' >> index.md;done`

This essentially allows to:
1. Create an index page of all your foam notes
2. Use any kind of filter(Filename patterns, Content RegEx patterns,...)
3. Ignore specific files/folders after a pattern,...

You could expand this and sort for the amount of foam links in it(Weight). But this was not in my scope :smile:  

---

Maybe it helps someone in the meantime.

**Comment by cliffordfajardo** [comment]
I tend to fiddle around with other PKM just to learn from their user interfaces and Reflect.app has a "All notes" tab which lists all your notes in a list like so:

![image](https://user-images.githubusercontent.com/6743796/156606466-9b1ddedc-fcae-48ef-a790-ba7291e05650.png)

Note: I don't use the app, so I cannot give detailed description about it; I'm currently using Foam and sometimes use Obsidian

Creating a custom UI for this type of view is possible; I haven't written a VSCode extension in about 2 years (_I've published a few tho_)
I started a discussion here about the new VSCode APIs/utility libraries that Microsoft & Github have been collaborating on here and how they might be relevant to Foam:
- https://github.com/foambubble/foam/discussions/951
